feature/DM-4962-opcua-gateway-java17 (#3812)
SoftwareAG/c8y-docs: content/change-logs/device-management/opcua-1022-0-0-opcua-gateway-internal-db.md

---
date:
title: OPC UA gateway now runs on Java 17
product_area: Device management & connectivity
change_type:
  - value: change-3BQrQ6adS

>     label: API change

Reviewer: @BeateRixen it's breaking change for the upgrade process and there is no change in the APIs.
Does it still makes sense to have this change type still or is "Announcement" type fits better here?

Reviewer 2: Breaking changes that do not affect the API should not use this label. I actually don´t like Announcement as this has already happened. I think we use Improvement in other cases. But Announcement might be better here.

Author: > Breaking changes that do not affect the API should not use this label. I actually don´t like Announcement as this has already happened. I think we use Improvement in other cases. But Announcement might be better here.

I didn't know what to choose here... I thought API change is best since we're changing Java version from 11 to 17 and there is no change type that matches this upgrade. Please decide with @Oezge-Akin what should be the change type and I will change it.

Reviewer 2: @jakubsobolewskisag Please use Announcement

Author: done

component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-4961
version: 1022.0.0
---
Together with the update to Java 17, the internal database of the OPC UA gateway has been changed from MapDB and ChronicleMap to RocksDB. 
This change enhances the performance and reliability of the gateway, ensuring compatibility with Java 17.

> **Caution:** Note that this update may require additional data migration or re-registration of device gateways. 

Reviewer 2: ```suggestion

{{< c8y-admon-req >}}
Note that this update may require additional data migration or re-registration of device gateways. 
```

> **Caution:** Please note that this update may require additional data migration or re-registration of device gateways. 

Reviewer 2: ```suggestion
**Caution:** Note that this update may require additional data migration or re-registration of device gateways. 
```

Reviewer 2: You better use the important admonition here.

Author: > important admonition

how to do it? Could you give an example?

> Please refer to [Upgrading from 2021 to 2022 gateway version](/device-integration/opcua/#upgrade-2021-to-2022) documentation section. 

Reviewer: ```suggestion
Please refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) documentation section. 
```

> Please refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) documentation section. 

Reviewer 2: ```suggestion
Refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) in the documentation. 
```

> It is recommended to back up your existing local gateway data before proceeding with the update.

Reviewer 2: ```suggestion
{{< /c8y-admon-important >}}

It is recommended to back up your existing local gateway data before proceeding with the update.
```

---
date:
title: OPC UA gateway now runs on Java 17
product_area: Device management & connectivity
change_type:
  - value: change-3BQrQ6adS
    label: API change
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-4961
version: 1022.0.0
---
Together with the update to Java 17, the internal database of the OPC UA gateway has been changed from MapDB and ChronicleMap to RocksDB. 
This change enhances the performance and reliability of the gateway, ensuring compatibility with Java 17.

> {{< c8y-admon-req >}}

Reviewer 2: ```suggestion
{{< c8y-admon-important >}}
```

---
date:
title: OPC UA gateway now runs on Java 17
product_area: Device management & connectivity
change_type:
  - value: change-3BQrQ6adS
    label: API change
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-4961
version: 1022.0.0
---
Together with the update to Java 17, the internal database of the OPC UA gateway has been changed from MapDB and ChronicleMap to RocksDB. 
This change enhances the performance and reliability of the gateway, ensuring compatibility with Java 17.

{{< c8y-admon-important >}}
Note that this update may require additional data migration or re-registration of device gateways. 

> Refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) in the documentation. 

Reviewer 2: Not sure, if this sentence is supposed to be part of the "important" information? Otherwise it must be moved below this box.

Reviewer 2: @jakubsobolewskisag  The first sentence was a note. But it did not have a line break. If the second sentence (reference to the docs) refers to the first sentence, then it can stay as is. If it is a general advice we should move it below the info box.

Author: @BeateRixen The second sentence refers to the first one. Should I change anything?

Author: @BeateRixen The second sentence refers to the first one. Should I change anything?

Reviewer 2: @jakubsobolewskisag How about this sentence? "It is recommended to back up your existing local gateway data before proceeding with the update." Should we move it into the "important" box as well?

> Refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) in the documentation. 

Reviewer 2: ```suggestion
Refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) in the documentation. 

It is recommended to back up your existing local gateway data before proceeding with the update.
```

---
date:

> title: OPC UA gateway now runs on Java 17

Reviewer 2: @jakubsobolewskisag The title should be adapted as it is currently the same as the one from the change log below.

Author: done

---
date:
title: OPC UA gateway now runs on Java 17
product_area: Device management & connectivity
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-4961
version: 1022.0.0
---
Together with the update to Java 17, the internal database of the OPC UA gateway has been changed from MapDB and ChronicleMap to RocksDB. 
This change enhances the performance and reliability of the gateway, ensuring compatibility with Java 17.

{{< c8y-admon-important >}}
Note that this update may require additional data migration or re-registration of device gateways. 
Refer to [Upgrading from 1021 to 1022 gateway version](/device-integration/opcua/#upgrade-1021-to-1022) in the documentation. 

It is recommended to back up your existing local gateway data before proceeding with the update.
{{< /c8y-admon-important >}}

> It is recommended to back up your existing local gateway data before proceeding with the update.

Reviewer 2: ```suggestion
```
